### PR TITLE
kpb: schedule draining task, fix issue in clearing history buffer and more

### PR DESCRIPTION
**Pull request by mrajwa**
In this patch draining task has been scheduled. 
Issue with wrong size calculation inside kpb_clear_history_buffer() has been fixed.
Added more logs + general cleanup.

### CODE REVIEW

File: src/audio/kpb.c [hunk 1]
```diff
@@ -384,9 +384,13 @@ static int kpb_prepare(struct comp_dev *dev)
...
 		if (sink->sink->comp.type == SOF_COMP_SELECTOR) {
-			/* we found proper sink */
+			/* we found proper real time sink */
 			cd->rt_sink = sink;
-			break;
+		} else if (sink->sink->comp.type == SOF_COMP_HOST) {
+			/* we found proper host sink */
+			cd->cli_sink = sink;
+		} else {
+			continue;
```
Comment (tlauda) [inline src/audio/kpb.c]: I don't think it's needed.

File: src/audio/kpb.c [hunk 2]
```diff
@@ -336,26 +336,28 @@ static int kpb_prepare(struct comp_dev *dev)
 	if (ret == COMP_STATUS_STATE_ALREADY_SET)
 		return PPL_STATUS_PATH_STOP;
 
+	/* Init private data */
 	cd->kpb_no_of_clients = 0;
+	cd->buffered_data = 0;
 
-	/* init history buffer */
+	/* Init history buffer */
```
Comment (lgirdwood) [inline src/audio/kpb.c]: These comment updates should be a separate patch.

File: src/audio/kpb.c (hunk 1 above)
Comment (lgirdwood) [reply src/audio/kpb.c]: @mrajwa why does KPB need to know where host is ? I dont understand why it has to understand the topology.

Comment (mrajwa) [reply src/audio/kpb.c]: @tlauda, why do you think it is not needed? How should we acquire sink address otherwise?
@lgirdwood, because we need to drain data somewhere right? So we need to get sink address. This is on KPB's sink list and as long as we work only with one client it is sufficient. With multiple of customers we need to introduce clients registration procedure.

File: src/audio/kpb.c (hunk 2 above)
Comment (mrajwa) [reply src/audio/kpb.c]: Done.

File: src/audio/kpb.c (hunk 1 above)
Comment (tlauda) [reply src/audio/kpb.c]: @mrajwa I mean continue label is not necessary.

Comment (mrajwa) [reply src/audio/kpb.c]: Ok, I agree the whole else condition is not necessary.
